decidim-core/lib/tasks/upgrade/clean_hidden_resources.rake

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :upgrade do
+    namespace :clean do
+      desc "Removes all related resources from hidden resource"
+      task hidden_resources: :environment do
+        logger.info("Removing child resources for hidden parents...")
+        Decidim::Moderation.hidden.find_each do |moderation_for_hidden_resource|
+          reportable = moderation_for_hidden_resource.reportable
+          current_user = reportable.organization.users.find_by!(email: Decidim::Ai::SpamDetection.reporting_user_email)
```

Comment:
This is adding a dependency to an optional module (`decidim-ai`) can you add a conditional for the module if it is available or not?

Reply:
fixed by [51081d2](https://github.com/decidim/decidim/pull/13554/commits/51081d23a2980e9f331207d52aa0f8f7071fc563)